packages/menu/src/MenuItem.ts

```diff
@@ -276,6 +330,110 @@ export class MenuItem extends LikeAnchor(Focusable) {
         }
     }
 
+    public closeOverlay?: (leave?: boolean) => Promise<void>;
+
+    protected handleSubMenuClick(): void {
+        this.openOverlay({ immediate: true });
+    }
+
+    protected handlePointerenter(): void {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        } else {
+            this.openOverlay();
+        }
+    }
+
+    protected leaveTimeout?: ReturnType<typeof setTimeout>;
+
+    protected handlePointerleave(): void {
+        if (this.hasSubMenu && this.open) {
+            this.active = true;
+            this.leaveTimeout = setTimeout(() => {
+                delete this.leaveTimeout;
+                if (this.closeOverlay) this.closeOverlay(true);
+            }, 100);
```

Comment:
should this timing (100) be defined as a CONST?

Reply:
I'm also not sure that `100` is the perfect number here, but it seemed like a good start when I stole it from your implementation. 😬